### PR TITLE
[fix] Do not create .htaccess with Nginx

### PR DESCRIPTION
**Pull request by magikcypress**
It's probably a good patch, but it is to discuss this issue #222

### CODE REVIEW

File: lib/Persistence/AbstractPersistence.php [hunk 1]
```diff
@@ -88,16 +88,18 @@ protected static function _initialize()
...
-        if (!is_file($file)) {
-            $writtenBytes = @file_put_contents(
-                $file,
-                'Require all denied' . PHP_EOL,
-                LOCK_EX
-            );
-            if ($writtenBytes === false || $writtenBytes < 19) {
-                throw new Exception('unable to write to file ' . $file, 11);
-            }
+        if ((strncmp("Apache", $_SERVER['SERVER_SOFTWARE'],6) == 0) or defined('_SERVER_APACHE')) {
```
Comment (rugk) [inline lib/Persistence/AbstractPersistence.php]: Is `_SERVER_APACHE` meant as a way to circumvent this check for admins? If so I'd say this should maybe better be a configuration variable… If not then is this really an official variable set by Apache?

Comment (magikcypress) [inline lib/Persistence/AbstractPersistence.php]: Thanks for you're welcome :+1: 

It's not official variable, this variable may call like this `define('_SERVER_APACHE',  'false');`

Yes, it seems interesting to put it in the configuration, but I do not know how to do it the best.

Comment (elrido) [inline lib/Persistence/AbstractPersistence.php]: The unit tests fail due to the missing check if the key SERVER_SOFTWARE exists. Also we need to add tests that mocks the existence of that variable emulating both an apache and non-apache environments to ensure the .htaccess files get created in Apache and not in others.

I can look into writing these tests later some time during this week.

Comment (rugk) [inline lib/Persistence/AbstractPersistence.php]: @elrido What do you say about the environmental variable in general? I'd say changing it to a config option is better…

@magikcypress If you want to see how to add a config option, in [this commit](https://github.com/PrivateBin/PrivateBin/commit/f54036976a486d3f7c83e52fd37ea357fd4e7a62) a config option was added.

**Comment by magikcypress** [comment]
Sorry, this PR is dirty ! 
The best is to create a new PR?

File: cfg/conf.ini.sample [hunk 2]
```diff
@@ -3,9 +3,6 @@
 ; An explanation of each setting can be find online at https://github.com/PrivateBin/PrivateBin/wiki/Configuration.
 
 [main]
-; (optional) set a project name to be displayed on the website
-; name = "PrivateBin"
-
```
Comment (rugk) [inline cfg/conf.ini.sample]: Please merge the current master into your project. You've accidentally reverted some commits/code changes we did in the master branch recently.

**Comment by rugk** [comment]
No, a quick fix could be to just run `git pull https://github.com/PrivateBin/PrivateBin` in the repo and it should merge the master into it.
Just keep your fingers crossed in order not to have to manually correct any conflicts… :wink:

**Comment by magikcypress** [comment]
My bad ! I could not do it ! Sorry, I tried to merge the master on patch-1 :( 

`git clone --branch patch-1 git@github.com:magikcypress/PrivateBin.git`
`git pull https://github.com/PrivateBin/PrivateBin`
`git merge master`

My merge doesn't work, and my changes are not made !?

You have a idea ?

**Comment by rugk** [comment]
After `git pull` the changes should already be merged. No "merge master" is necessary afterwards anymore. Usually a `git push` afterwards should publish the changes.

**Comment by magikcypress** [comment]
I managed to catch up with my mistakes ?!

**Comment by rugk** [comment]
Thanks for these changing (especially fixing the indentation), but it seems your changes still revert some commits in the master branch…

Either follow [this guide](https://help.github.com/articles/merging-an-upstream-repository-into-your-fork/) again or – if it is easier for you – create a new PR.

**Comment by magikcypress** [comment]
I have created a new PR tomorrow. 
Github is my worst enemy at the moment ;)

**Comment by rugk** [comment]
:smile:
Yeah, no problem. You'll learn it.

**Comment by rugk** [comment]
BTW when creating a new PR you can add a string in order to (automatically) [let issues close](https://help.github.com/articles/closing-issues-via-commit-messages/) when a PR is merged.

**Pull request closed without merging**